fullscreen/model/remove-first-manual.html

```diff
@@ -23,6 +23,9 @@
             assert_equals(document.fullscreenElement, last);
             assert_equals(event.target, last);
             first.remove();
+            // Both /first/ and /last/ were removed from the top layer and
+            // thus the fullscreen element synchronously becomes null.
+            assert_equals(document.fullscreenElement, null);
```

Comment:
So in Firefox, when any of the fullscreen element is unbound from the tree, we exit fullscreen and queue events synchronously, then force the window to restore asynchronously (as it needs to be).

I guess either way is fine to me, although I'm not very motivated to change code here, because removing fullscreen element should be considered as an error case, so the bottom line is we should exit fullscreen safely, and I don't think we should expect any legitimate content to rely on behavior here.